### PR TITLE
Fix table spacing, fill and cursor

### PR DESCRIPTION
**Pull request by mayank99**
- Added margin-right to filter icon
- Added iui-actionable class for cursor: pointer in header cells
- Added iui-icons-color fill for sort icon hover state
- Formatting fixes

### CODE REVIEW

File: src/table/table.scss [hunk 1]
```diff
@@ -36,10 +36,14 @@
...
     @include iui-focus;
-    cursor: pointer;
+
+    &.iui-actionable {
+      cursor: pointer;
+    }
 
     > .iui-filter-button {
       margin-left: $iui-xs;
+      margin-right: $iui-xs;
```
Comment (mayank99) [inline src/table/table.scss:46]: @FlyersPh9 Can you check what the value here should be?

0 (current):
image

xs (4px):
image

s (8px)
image

Comment (FlyersPh9) [reply src/table/table.scss:46]: My initial thought is to have the spacing between the button and the right edge of the column to be equal to either the red or orange spacing in the image below.
Screen Shot 2021-06-14 at 11 45 53 AM

I know the red gap is 16px, which is probably not our best use of space.  And the orange gap is probably too close to the edge of the column and a user might accidentally hover over the column resizer.  So I'd say let's go with the 8px gap.

File: src/table/table.scss [hunk 2]
```diff
@@ -99,6 +103,7 @@
     visibility: hidden;
     @include themed {
       fill: t(iui-icons-color-actionable);
+      opacity: t(iui-opacity-3);
```
Comment (FlyersPh9) [inline src/table/table.scss:106]: Another option could be to use `iui-icons-color` for the hover state & `iui-icons-color-actionable` for the active sort state.

Comment (mayank99) [reply src/table/table.scss]: Done

File: src/table/table.scss (hunk 1 above)
Comment (mayank99) [reply src/table/table.scss]: Changed to `iui-s`

**Review by FlyersPh9 — APPROVED** [review]

**Pull request merged**